dashboard/api.py

```diff
@@ -479,10 +482,16 @@ def get_certificate_url(mmtrack, course):
     Returns:
         str: url to view the certificate
     """
-    final_grades = mmtrack.get_passing_final_grades_for_course(course)
     url = ""
+    final_grades = mmtrack.get_passing_final_grades_for_course(course)
     if final_grades.exists():
         best_grade = final_grades.first()
-        if best_grade.has_certificate:
-            url = "/certificate/{}".format(final_grades.first().certificate.hash)
+        course_key = best_grade.course_run.edx_course_key
+        if mmtrack.financial_aid_available:
+            if best_grade.has_certificate:
+                url = reverse('certificate', args=[final_grades.first().certificate.hash])
```

Comment:
isn't `final_grades.first()` equal to `best_grade`?

Reply:
yes, it is